src/NuGet.Services.Metadata.Catalog.Monitoring/Resources/PackageRegistrationMetadataResourceV3.cs

```diff
@@ -24,33 +25,54 @@ public class PackageRegistrationMetadataResourceV3 : IPackageRegistrationMetadat
             _client = client;
         }
 
-        public async Task<PackageRegistrationIndexMetadata> GetIndex(PackageIdentity package, ILogger log, CancellationToken token)
+        public async Task<PackageRegistrationIndexMetadata> GetIndexAsync(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromIndex(package, log, token);
-            return feedPackage != null ? 
-                JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) : 
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromIndex(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
+                throw new ValidationException($"Could not fetch {nameof(PackageRegistrationIndexMetadata)} from V3!", e);
```

Comment:
Because this makes it very clear that the issue is in the parsing of registration pages. The exception that is thrown by `Newtonsoft.Json` is confusing in our context. Wrapping the exception will reduce debugging time.